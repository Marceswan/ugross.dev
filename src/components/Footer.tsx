import React from 'react';
import { NAV_ITEMS } from '~/configs/site.config';
import Link from 'next/link';

interface FooterProps {
  copyrightText: string;
}

export const Footer: React.FC<FooterProps> = ({ copyrightText }) => {
  return (
    <footer className="text-muted text-sm text-center mt-auto pt-20">
      <div className="container flex flex-col space-y-5 sm:space-y-0 sm:flex-row items-center justify-between border-t dark:border-gray-700 py-6 px-2">
        <div className="flex space-x-5">
          {NAV_ITEMS.map(({ url, name }) => (
            <Link href={url} key={url}>
              <a className="font-normal border-0 hover:text-link transition-colors">{name}</a>
            </Link>
          ))}
        </div>
        <div>
          <p className="m-xs">{copyrightText}</p>
        </div>
      </div>
    </footer>
  );
};
