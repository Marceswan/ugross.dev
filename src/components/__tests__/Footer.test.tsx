import { mount, shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Footer from '../Footer';

describe('Footer component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render footer HTML element', () => {
    const wrapper = mount(<Footer />);
    expect(wrapper.find('footer').length).toBe(1);
  });

  it('should contain actual year in the copyright text', () => {
    const wrapper = mount(<Footer />);
    const currentYear = new Date().getFullYear();
    expect(
      wrapper
        .find('p')
        .html()
        .indexOf(currentYear.toString())
    ).toBeTruthy();
  });
});
