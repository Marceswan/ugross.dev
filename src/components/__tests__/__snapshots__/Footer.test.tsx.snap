// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<footer
  className="sc-bdVaJa dMHGVI"
>
  <p>
    Â© Copyright 
    2018
     Rostyslav Ugryniuk
  </p>
</footer>
`;
